python/pyarrow/_parquet.pyx

```diff
@@ -1941,6 +1943,14 @@ cdef shared_ptr[WriterProperties] _create_writer_properties(
                     "'use_byte_stream_split' cannot be passed"
                     "together with 'column_encoding'")
 
+    # store_decimal_as_integer
+
+    if isinstance(store_decimal_as_integer, bool):
```

Comment:
Do we expect `store_decimal_as_integer` to ever not be a list of columns like `use_byte_stream_split` can be above? If not we don't need this check.

Reply:
At present, I believe the C++ method does not accept columns.

Reply:
In that case do we need to check if it's a boolean or not?

Reply:
What I'm trying to say is we can probably remove the `if isinstance(store_decimal_as_integer, bool):` check if we don't need `store_decimal_as_integer` to ever be non-boolean.

Reply:
If we expect this to be always a boolean, in my opinion, we probably should raise a `ValueError` if it is not a boolean, as we do with other options, example `column_encoding` :
```
        else:
            raise TypeError(
                "'column_encoding' should be a dictionary or a string")
```
instead of silently ignoring it.

Reply:
As this is if `store_decimal_as_integer` is non-boolean it will get ignored. We could raise in case of other types to ensure hygiene?
Also isn't one of the two behaviors default? Why don't we just set in case we're using the non default one?
```
        if store_decimal_as_integer:
            props.enable_store_decimal_as_integer()
        else:
            props.disable_store_decimal_as_integer()
```

Reply:
@rok is right, ignoring arguments like this is very bad.

Reply:
I added an `else` clause which raises a `TypeError`. 

I did not add a default props.*able_store_decimal_as_integer() as I thought it would be better to defer to the C++ writer defaults, which I believe is to disable.

Reply:
>  I believe its better to be explicit rather than implicit here in the if/else with the property

That's a fair point - if c++ defaults change it would surprise python users.